pygments/formatters/latex.py

```diff
@@ -129,6 +133,7 @@ def escape_tex(text, commandprefix):
 \def\%(cp)sZlb{[}
 \def\%(cp)sZrb{]}
 \makeatother
+\%(cp)s@restore@catcode@of@minus@sign
```

Comment:
as commented above we could also decide to go for only ``\catcode`\-=12 `` approach because nobody in his/her right mind should input a TeX/LaTeX file of macros with any other catcode for `-` than 12